gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+
+  const { classList } = document.documentElement;
+
+  useEffect(() => {
+    classList.toggle('nx-html--enable-color-schemes', enableModeChange);
+  }, [enableModeChange]);
+
+  useEffect(() => {
+    classList.toggle('nx-html--dark-mode', mode === 'dark' && enableModeChange);
+    classList.toggle('nx-html--light-mode', mode === 'light' && enableModeChange);
+  }, [enableModeChange, mode]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Theme Settings</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 onCancel={modalCloseHandler}
+                 aria-labelledby="nx-modal-dark-mode-form-header">
+          <NxForm onSubmit={modalCloseHandler}
+                  submitBtnText= "Save and Close">
+            <header className="nx-modal-header">
+              <h2 className="nx-h2" id="nx-modal-dark-mode-form-header">
+                Setting Preferences for Light or Dark Mode
+              </h2>
+            </header>
+            <div className="nx-modal-content">
+              <NxFormGroup label="Enable Theme Changes">
+                <NxToggle onChange={toggleEnableModeChange} isChecked={enableModeChange}>
+                  Opt-in to Allow Theming
+                  <NxTooltip title="Opting in will allow you to make adjustments to the display theme. Failure
+                             to do so will always result in light mode. This can be done by adding the
+                             'nx-html--enable-color-schemes' class to the HTML element.">
```

Comment:
Note that tooltip titles can be JSX.  I think it would be worth trying `NxCode` where appropriate within the tooltip to see how it looks.

Reply:
> Note that tooltip titles can be JSX. I think it would be worth trying `NxCode` where appropriate within the tooltip to see how it looks.

 I did try it initially, but wasn't about it's appearance. Here's a screenshot of what it looks like inside the tooltip:
![Screen Shot 2022-09-22 at 8 49 54 AM](https://user-images.githubusercontent.com/89862727/191754650-f8c0bbaf-50d4-4fe5-a9d3-415807cf5c6f.png)

Reply:
> I did try it initially, but wasn't sure about if it looked ok. Here's a screenshot of what it looks like inside the tooltip:

Check with design